src/Nethermind/Nethermind.Consensus.AuRa/Transactions/PartiallyEagerTxSource.cs

```diff
@@ -0,0 +1,97 @@
+﻿//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+using System.Collections.Generic;
+using Nethermind.Consensus.Transactions;
+using Nethermind.Core;
+
+namespace Nethermind.Consensus.AuRa.Transactions
+{
+    public class PartiallyEagerTxSource : ITxSource, IDisposable
+    {
+        private readonly ITxSource _innerSource;
+        private readonly Predicate<Transaction> _eagerTransaction;
+        private IEnumerator<Transaction>? _enumerator;
+        private readonly IList<Transaction> _eagerTransactions = new List<Transaction>();
```

Comment:
what are eager transactions? can we name this in a way that tells us what they do?